test/e2e/framework/unmanaged.go

```diff
@@ -95,28 +97,33 @@ func (f *UnmanagedFramework) AfterEach() {
 
 	userAgent := fmt.Sprintf("%s-teardown", f.BaseName)
 
-	// DeleteNamespace at the very end in defer, to avoid any
-	// expectation failures preventing deleting the namespace.
-	defer func() {
-		if f.testNamespaceName == "" {
-			return
-		}
-		// Clear the name first to ensure other tests always get a
-		// fresh namespace even if namespace deletion fails
-		namespaceName := f.testNamespaceName
-		f.testNamespaceName = ""
-
-		client := f.KubeClient(userAgent)
-		deleteNamespace(client, namespaceName)
-	}()
-
-	// Print events if the test failed and ran in a namep.
+	// Print events if the test failed and ran in a namespace.
 	if CurrentGinkgoTestDescription().Failed && f.testNamespaceName != "" {
 		kubeClient := f.KubeClient(userAgent)
 		DumpEventsInNamespace(func(opts metav1.ListOptions, ns string) (*corev1.EventList, error) {
 			return kubeClient.Core().Events(ns).List(opts)
 		}, f.testNamespaceName)
 	}
+
+	// DeleteNamespace at the very end but before tearing down the namespace
```

Comment:
Cleanup needs to remain as a defer function because `DumpEventsInNamespace` contains assertions that could exit the function.

Reply:
Ah, interesting. I added a comment about that and moved the original code, plus the fixture cleanup which depends on #48, to a defer function.